plutus-ledger-constraints/src/Ledger/Constraints/OnChain/V2.hs

```diff
@@ -153,12 +153,12 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
                 -- provide datum.
                    Ada.fromValue txOutValue >= Ada.fromValue vl
                 && Ada.fromValue txOutValue <= Ada.fromValue vl + Ledger.maxMinAdaTxOut
-                && Value.noAdaValue txOutValue == Value.noAdaValue vl
+                && geq (Value.noAdaValue txOutValue) (Value.noAdaValue vl)
```

Comment:
What is the logic for this instead of `==`?

Reply:
The same as for ada values, we want to have at least the requested amounts of each requested values available.

Reply:
Mmm... but for the Ada values the only reason we did that was because of the min Ada per utxo requirement. I thought this wouldn't be needed for non-Ada values. Maybe I need to check the test case to understand.

Reply:
I had a similar thought. The most important thing is that MustMintValue is strict with the amount, which it is.

Reply:
As noticed during the review with James, the good thing is that the new semantic for `MustPayToOtherScript` onchain is now similar to the one of the `MusPayToPubKey` constraint.